paasta_tools/vitesscluster_tools.py

```diff
@@ -1,31 +1,581 @@
+        "db-credentials-server": "vault",
+        "db-credentials-vault-addr": f"https://vault-dre.{region}.yelpcorp.com:8200",
+        "db-credentials-vault-path": "secrets/vitess/vt-tablet/vttablet_credentials.json",
+        "db-credentials-vault-tls-ca": f"/etc/vault/all_cas/acm-privateca-{region}.crt",
+        "db-credentials-vault-ttl": "60s",
+    }
+    vttablet_extra_flags.update(flag_overrides)
+
+    environment_overrides: Dict[str, Any] = {
+        "VAULT_ADDR": f"https://vault-dre.{region}.yelpcorp.com:8200",
+        "VAULT_CACERT": f"/etc/vault/all_cas/acm-privateca-{region}.crt",
+        "TOPOLOGY_FLAGS": f"--topo_implementation {TOPO_IMPLEMENTATION} --topo_global_server_address ${zk_address} --topo_global_root {TOPO_GLOBAL_ROOT}",
+        "CELL_TOPOLOGY_SERVERS": zk_address,
+        "DB": db_name,
+        "KEYSPACE": keyspace,
+    }
+    environment_overrides.update(VTTABLET_EXTRA_ENV)
+    updated_vttablet_extra_env = (
+        get_updated_environment_variables(environment_overrides) + env
+    )
+
+    # Add extra pod label to filter
+    tablet_type_label = limit_size_with_hash(name=f"{db_name}_{tablet_type}", limit=63)
+    labels.update({"tablet_type": tablet_type_label})
+
+    try:
+        type = load_system_paasta_config().get_vitess_tablet_pool_type_mapping()[
+            tablet_type
+        ]
+    except KeyError:
+        log.error(
+            f"Tablet type {tablet_type} not found in system paasta config vitess_tablet_pool_type_mapping"
+        )
+
+    try:
+        replicas = int(vttablet_resources.get("replicas"))
+    except ValueError:
+        log.error("Invalid replicas value for vttablet")
+        replicas = 1
+    requests = vttablet_resources.get("requests", {"cpu": "100m", "memory": "256Mi"})
+
+    config = TabletPoolDict(
+        cell=cell,
+        name=f"{db_name}_{tablet_type}",
+        type=type,
+        affinity={"nodeAffinity": node_affinity},
+        extraLabels=labels,
+        extraEnv=updated_vttablet_extra_env,
+        extraVolumeMounts=[
+            {
+                "mountPath": "/etc/vault/all_cas",
+                "name": "vault-secrets",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "/etc/vitess_keyspace_acls",
+                "name": "acls",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "etc/credentials.yaml",
+                "name": "vttablet-fake-credentials",
+                "readOnly": True,
+            },
+            {
+                "mountPath": "/etc/init_db.sql",
+                "name": "keyspace-fake-init-script",
+                "readOnly": True,
+            },
+        ],
+        extraVolumes=[
+            {"name": "vault-secrets", "hostPath": {"path": "/nail/etc/vault/all_cas"}},
+            {
+                "name": "acls",
+                "hostPath": {"path": "/nail/srv/configs/vitess_keyspace_acls"},
```

Comment:
hmm, almost missed this - this is probably going to break in surprising ways later on: `/nail/srv/configs` is a symlink that will have its target change pretty frequently (every srv-configs deployment)

```
[I] dev208-uswest1adevc:~> exa -la /nail/srv/                                                                                                                                                                                                                  
lrwxrwxrwx 26 root 26 Aug  2022 configs -> /nail/etc/srv-configs/main

[I] dev208-uswest1adevc:~> exa -la /nail/etc/srv-configs/main                                                                                                                                                                                                   
lrwxrwxrwx 49 root 13 Jun 15:07 /nail/etc/srv-configs/main -> /nail/etc/srv-configs/.client/../.main-1718316455
```

Reply:
what we normally do here is that we mount `/nail/srv` (which is not a symlink) and thus avoid losing track of files/dirs when the inodes go away :)

which brings me to another question: i wonder if it's worth using the code that we use for services to add in the puppet-configured default paasta volume mounts (which includes easy-to-get-wrong mounts like the srv-configs one)

Reply:
Fixed in https://github.com/Yelp/paasta/pull/3889/commits/2f6ecc76417760dbd1cb424c3b61dabd5c83ea09. Lmk if this looks good now